### ISSUE TITLE
Support archiving

### ISSUE THREAD

**Issue by pbx** [orgzly/orgzly-android#32]
Org mode has this handy feature, archiving, that lets you move an item or subtree to a separate archive file. This would be handy as an option in the swipe menus (along with state change, moving, deleting, etc.).

I recognize this would have some complexity, since it involves writing to a second file.

http://orgmode.org/manual/Moving-subtrees.html#Moving-subtrees

**Comment by alphapapa** [comment]
Personally, I'd almost prefer if Orgzly didn't support this.  There are so many options and that control how archiving is done, Orgzly would have to support and test all of them, and it's not something I do often.  I'd definitely want an undo option...  :)

**Comment by mkaito** [comment]
Quoting alphapapa (2017-03-09 15:09:58)
> Personally, I'd almost prefer if Orgzly didn't support this. There are so many
> options and that control how archiving is done, Orgzly would have to support
> and test all of them, and it's not something I do often. I'd definitely want an
> undo option... :)

What's the scope of orgzly again? Should it be a *complement* to
org-mode, or a standalone program compatible with org-mode? Personally,
I'd prefer the former, although I can see the appeal of the latter.

With that in mind, I'd rather see orgzly focus on making my org-mode
files conveniently available on the go rather than reimplementing all of
org-mode. Archival is one of those things that I'd rather do on the
computer anyway.

**Comment by danShumway** [comment]
Agreed, issues like #15 would be more useful for me. I think of orgzly as a convenient interface/view into org-mode, not as a full reimplementation.

I don't even make my archives available on my phone. I mark something as done, and then when I get to my computer I archive all my done items. I can imagine a world where someone would find it useful, but if orgzly becomes a full reimplementation of org-mode there are tradeoffs there as well. 

I don't want to minimize anyone or their workflow, I'm just curious how much demand there actually is.

**Comment by nevenz** [comment]
This shouldn't be too hard to implement. Moving is easy and all properties should be available (though some a bit slow to grab currently, like ARCHIVE_ITAGS, though for a single-note action it's not a big deal).

Not a very high priority to me though, but great task for getting to know the code. ;)

**Comment by alphapapa** [comment]
I would just want to make sure that, if Orgzly does support archiving, it does it in *exactly* the same way full Org-mode does it.

**Comment by erikstokes** [comment]
It's probably much simpler to implement [Internal Archiving](http://orgmode.org/manual/Internal-archiving.html#Internal-archiving) since this just involves adding an `ARCHIVE` tag and knowing to hide such entries.  My archiving setup is complex enough that orgzy probably won't do it just the way I like, but adding a tag is just adding a tag.

**Comment by vibhavsinha** [comment]
As somebody who uses Orgzly solely on mobile and as a text file on computer without emacs, this feature will be super useful. At least _there should be an option to hide DONE items when viewing a notebook_.
Orgzly without org mode emacs is still better GTD app than most others.

**Comment by lebarde** [comment]
One more plus here. It is now the missing feature that prevents me from moving from Simpletask (todo.txt) to orgzly.
Actually I used to use org-mode on emacs  for notes only.

Now I use Simpletask to manage all my todo tasks everyday. I would need nested sections and other things (which is why I would like to use orgzly intensively) but most of all I need archiving. I don't go everyday on emacs, but I intensively use my productivity app on my smartphone.

**Comment by falematte** [comment]
I would love this feature too!

**Comment by mloetsch** [comment]
i would also love to hide items in the tree. either with a tag or/and items marked as done. this would be enough as i can always manually archive these items later on my pc

**Comment by Silverhawk85** [comment]
I would also find the feature handy. Org files can quickly become confusing if you don't archive or delete completed tasks from time to time. Especially archiving here means nothing else than moving a subtree to another file (for example standard to *.org_archive).

TODO.txt and as already mentioned the app Simpletask use such a feature, even with auto archiving once a task has been marked completed.
